seekinternational/notification-library
# Feature/EL-3243 Fix errors swallowed and undefined result

What does this PR do?
---------------------
We encounter 2 problems in one of Sonic projects where:

- Errors thrown inside notification library are swallowed which means we are unable to track properly in the consumer.
- Undefined result being returned instead of status code and body.

From my investigation, it is due to binding on async function. So far changing it to curry function works for me.

This PR is to:

- Fix the issues.
- Make Segment tracking optional.
- Add git hooks (husky) to make development easy.

How can I test this?
--------------------
1. Import the latest commit of this branch into any consumer.
2. Test the functions and ensure they are working fine.

Notes
-----
Kindly provide any other suggestion if possible.
